Firestore/core/test/firebase/firestore/testutil/testutil.cc

```diff
@@ -38,36 +43,34 @@ DocumentSet DocSet(DocumentComparator comp, std::vector<Document> docs) {
   return set;
 }
 
-std::unique_ptr<model::PatchMutation> PatchMutation(
+model::PatchMutation PatchMutation(
     absl::string_view path,
-    const model::FieldValue::Map& values,
+    model::FieldValue::Map values,
     // TODO(rsgowman): Investigate changing update_mask to a set.
-    const std::vector<model::FieldPath>* update_mask) {
-  model::ObjectValue object_value = model::ObjectValue::Empty();
-  std::set<model::FieldPath> object_mask;
+    std::vector<model::FieldPath> update_mask) {
+  bool merge = !update_mask.empty();
```

Comment:
Nit: what's the reason to move the initialization of `merge` farther away from the place where it's used?

Reply:
It was to make it more closely match the Objective-C test helper. However, looking at Android keeping it where it was would have been better. Reverted.